### DISCUSSION TITLE
requirements.txt request

### DISCUSSION THREAD

**Discussion by benfwalla** [microsoft/ML-For-Beginners#676] [Ideas]
Would love to get a `requirements.txt` file that shows all the python libraries that were used in the creation of of this awesome curriculum. I ran into a couple scikit and seaborn issues here and there that were solved by updating the code to reflect the version I downloaded.

Giving students a requirements.txt to download with `pip` would help solve a lot of these runtime problems

**Comment by Hamza-farid** [comment]
That’s a great suggestion. Having a requirements.txt (or even a pyproject.toml/environment.yml for Conda users) makes it much easier to ensure everyone is running the same versions of the key libraries.

If one isn’t provided, you can also generate it yourself from your working environment using:

pip freeze > requirements.txt


This will capture all installed packages and their versions. Sharing that file with others (or asking contributors to add an official one to the repo) would definitely help avoid version conflicts, especially with libraries like scikit-learn and seaborn that change APIs across releases.
